### ISSUE TITLE
Helper utilities for Editor Scripts

### ISSUE THREAD

**Issue by koenbollen** [defold/defold#9458]
**Is your feature request related to a problem? Please describe (REQUIRED):**
I had a usecase the other day I couldn't fulfill with an editor-script (and made a pythonscript instead).

I wanted to make an action that lists .wav files in a specific directory. Generate a .go files based on those files and write out that file. The problem I ran into was listing files. A helper function for that would be awesome (especially crossplatform).

**Describe the solution you'd like (REQUIRED):**
A collection of util/helper functions, especially for cross-platform-tricky tasks (like the difference between `dir` and `ls`).

My ideas (I'll update based on comments):
- **listing files in a directory** - Maybe based on a glob pattern. e.g:  
  ```lua
  local wavFiles = editor.list_files("assets/sfx/*.wav")
  ```
- **creating a game object with a list of components** - Or edit a GO, like adding sound components
- **reading/writing files** - all tho I guess this can be done with basic lua functions.

**Describe alternatives you've considered (REQUIRED):**
I tried to do it with shell-outs and use `dir` and `ls`, but gave up and made a python script instead.

**Comment by HalfstarDev** [comment]
That would be great. I made a Lua module that can do some of that already, but a native implementation with more features would be better.

You could use [dir.lua](https://github.com/HalfstarDev/defold-index/blob/main/index/dir.lua) from the [defold-index](https://github.com/HalfstarDev/defold-index) extension and use:
```lua
local files = dir.get_list_files("assets/sfx")
```
and then filter for *.wav yourself.
It works on Windows and Linux, but I haven't tested it on macOS. But it works both from editor scripts and game scripts.

**Comment by ekharkunov** [comment]
Some similar thoughts described here https://github.com/defold/defold/issues/8504

**Comment by megalanthus** [comment]
> * **listing files in a directory** - Maybe based on a glob pattern. e.g:
>   local wavFiles = editor.list_files("assets/sfx/*.wav")

I support the idea, since it is currently not possible to get a list of files in a simple way, see: https://github.com/defold/defold/issues/8641#issuecomment-2763318005

**Comment by AGulev** [comment]
# Issue #9458: Helper utilities for Editor Scripts

https://github.com/defold/defold/issues/9458

Short summary: The original blocker in this feature request was cross-platform file listing from editor scripts so a script could enumerate files like `/assets/sfx/*.wav` and generate resources from that list. In the current codebase, editor scripts can already treat project directories as readable nodes and list their children, plus they now have dedicated resource and file helper APIs.

## Code areas inspected

- `editor/src/clj/editor/editor_extensions/graph.clj:101` - folder resource paths are resolved as folder resources, not rejected outright.
- `editor/src/clj/editor/editor_extensions/graph.clj:461` - `editor.get()` on a folder supports `"path"` and `"children"`, with `"children"` returning project paths for child resources.
- `editor/src/clj/editor/editor_extensions/graph.clj:489` - readable properties for folder resources are explicitly `["children" "path"]`.
- `editor/src/clj/editor/editor_extensions.clj:937` - the editor script runtime registers helper APIs including `create_directory`, `create_resources`, `delete_directory`, `resource_attributes`, `external_file_attributes`, and `execute`.
- `editor/src/clj/editor/editor_extensions/docs.clj:151` - these helper APIs are documented as public editor script functions.
- `editor/test/resources/editor_extensions/resources_as_nodes_project/test.editor_script:20` - test script uses `editor.get("/assets", "children")` to enumerate files in a project directory.
- `editor/test/integration/editor_extensions_test.clj:2081` - integration test asserts `editor.can_get("/assets", "children") == true` and verifies returned children include `/assets/a.png` and `/assets/b.png`.
- `editor/test/resources/editor_extensions/resources_io_project/test.editor_script:15` - test script uses `editor.create_resources()` to generate files and resources programmatically.

## Final assessment

Possibly outdated / fixed

## Rationale

The issue's main practical need appears to be already covered by the current editor script API:

- Project directories are first-class readable targets for editor scripts, and `editor.get("/some/folder", "children")` returns their contents.
- This is not just incidental behavior; it is implemented directly in `graph.clj` and covered by an integration test.
- That means the original example can now be implemented without shelling out to `dir` or `ls`: list `editor.get("/assets/sfx", "children")`, filter for `.wav`, then generate output.
- The runtime also now exposes additional helper APIs that align with the broader request for editor-script utilities: `editor.resource_attributes()`, `editor.external_file_attributes()`, `editor.create_directory()`, `editor.create_resources()`, and `editor.delete_directory()`.
- What is still missing is the exact higher-level convenience API proposed in the issue (`editor.list_files("assets/sfx/*.wav")`) and possibly a broader "utility collection" abstraction. But the concrete blocked workflow described in the issue is no longer blocked by platform differences.

Because the exact requested API shape was not added, but the underlying capability now exists and is tested, this looks more like an issue that has become stale or outdated than one that is still clearly actionable in its original form.

## Why this likely became outdated

- The original request was motivated by not being able to enumerate project files portably from editor scripts.
- Current editor-script APIs already provide a portable, repository-aware way to do that.
- Integration tests show that folder children enumeration is supported intentionally rather than accidentally.
- The newer helper APIs around creating directories and resources further reduce the gap between the requested workflow and what is already possible.

That evidence is strong enough to treat the issue as likely outdated even though no exact `editor.list_files()` helper was added.
